### PR TITLE
[chore] update to torch v1.7.0

### PR DESCRIPTION
**Pull request by msbaines**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/nn/pipe_process/test_pipe.py [hunk 1]
```diff
@@ -635,7 +635,7 @@ def deferred_batch_norm_params(checkpoint, lazy):
     assert torch.allclose(pipe[0].bias.grad, bn.bias.grad, atol=1e-4)
 
 
-@torch_spawn([5])
+@torch_spawn([4])
```
Comment (froody) [inline tests/nn/pipe_process/test_pipe.py:638]: why did you need this? was it causing an error? you should also change line 649 to be `if model.group.rank() == 3:`

**Review by froody — APPROVED** [review]

File: tests/nn/pipe_process/test_pipe.py (hunk 1 above)
Comment (msbaines) [reply tests/nn/pipe_process/test_pipe.py:638]: NCCL 2.7.8 does not support two processes having a context to the same GPU. Our circleci GPU instance only has 4 GPUs.
